### PR TITLE
Add QC outputs

### LINKED ISSUE #54: QC rules
Add some rules to perform QC for the workflow. QCs may involve the following:
- [x] Accuracy of segmentations (e.g. approproriate coverage of structures)

### PR DESCRIPTION
**Pull request by kaitj**
## Proposed changes

First pass at resolving #54, adding in both PNG and HTML outputs to QC the segmentation, as well the registration between the "template" and subject T1w images. The PNGs are mosaics with 7 cuts per view, while the HTML is interactive. 
It does use the 'viridis' colour map by default as their discrete colour maps doesn't seem like it can support more than 20 colours. I've tested it on some existing data on Graham. As for use, it's simply running the command (as one would to call the workflow), and the QC will be generated automatically.

Some nice to have long-term stuff that should be addressed in later PRs
* All of the QC files in a single HTML report
* Custom colour map to support more than 20 discrete colours

Since this is based off of @arunthurai's qc branch (and consequently the main branch?), documentation (#52) may need to be updated to include some information about QC and additional conditional logic (#58) may need to be implemented, dependent on the order of merging.

## Types of changes

What types of changes does your code introduce? Put an `x` in the boxes that apply

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other (if none of the other choices apply)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you are unsure about any of the choices, don't hesitate to ask!

- [x] Changes have been tested to ensure that fix is effective or that a feature works.
- [x] Changes pass the unit tests
- [x] Code has been run through the `poe quality` task
- [x] I have included necessary documentation or comments (as necessary)
- [x] Any dependent changes have been merged and published

## Notes
All PRs will undergo the unit testing before being reviewed. You may be requested to explain or make additional changes before the PR is accepted.

### CODE REVIEW

File: scattr/workflow/rules/qc.smk [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        config["singularity"]["scattr"]
+    script:
+        "../scripts/qc/registration_qc.py"
+
+
+rule gather_qc:
+    input:
+        dseg_png=expand(
+            rules.segment_qc.output.qc_png,
+            subject=config["input_lists"]["T1w"]["subject"],
```
Comment (tkkuehn) [inline scattr/workflow/rules/qc.smk:75]: Is there any chance these files will have `session`s?

**Review by tkkuehn — APPROVED** [review]
Looks like a good, mergeable to me, with the note that it would be good to post an example of what each type of QC image looks like here (for @ataha24 to comment on in particular). One note on dataset compatibility.

File: scattr/workflow/rules/qc.smk (hunk 1 above)
Comment (kaitj) [reply scattr/workflow/rules/qc.smk:75]: It is entirely possible - though I think that that is just a problem throughout the workflow that would be better saved to tackle in a separate PR.

**Comment by kaitj** [comment]
> Looks like a good, mergeable to me, with the note that it would be good to post an example of what each type of QC image looks like here (for @ataha24 to comment on in particular). One note on dataset compatibility. 

Ahh yeah, that is a good point - I'll pull those PNGs out of the zip in the morning and add it to this (as well as share the html files on slack).

**Comment by tkkuehn** [comment]
Needs a merge from main to pass the tests

**Comment by kaitj** [comment]
Just rebased from main - not sure why it didn't trigger the test though. Also still have a poetry.lock conflict...

**Comment by tkkuehn** [comment]
Yep fine by me pending any feedback from @ataha24. The `poetry.lock` conflict can be resolved by merging from main and just regenerating it with `poetry.lock`.

**Comment by kaitj** [comment]
Ahh, I think I understand - do you mean something like a gif that goes between the two? In, which case I do have some code that does that, though I would be hesitant to add that into this workflow, depending on how we ultimately want the "QC report" to look, as it involves some additional code. 

In any case, I will merge this in, and we can discuss more about the panning later on.

**Pull request merged**